src/test/java/org/beanpod/switchboard/dao/StreamDaoImplTest.java

```diff
@@ -88,6 +91,121 @@ public void testCreateChannel() {
     verify(streamRepository).save(streamEntity);
   }
 
+  @Test
+  public void testCreateChannelAlreadyExists() {
+    // given
+    CreateStreamRequest createStreamRequest = StreamFixture.getCreateStreamRequest();
+
+    when(streamRepository.existsDuplicate(
+            createStreamRequest.getInputChannelId(), createStreamRequest.getOutputChannelId()))
+        .thenReturn(true);
+
+    // when & then
+    assertThrows(
+        ExceptionType.StreamAlreadyExistsException.class,
+        () -> streamService.createStream(createStreamRequest));
+  }
+
+  @Test
+  public void testCreateChannelDifferentPublicIp() {
+    String otherPublicIpAddress = "179.256.54.21";
+    CreateStreamRequest createStreamRequest = StreamFixture.getCreateStreamRequest();
+    InputChannelDTO inputChannelDto = ChannelFixture.getInputChannelDto();
+    OutputChannelDTO outputChannelDto = ChannelFixture.getOutputChannelDto();
+    outputChannelDto.getEncoder().getDevice().setPublicIpAddress(otherPublicIpAddress);
+    StreamEntity streamEntity = StreamFixture.getStreamEntity();
+
+    when(channelService.getInputChannelById(createStreamRequest.getInputChannelId()))
+        .thenReturn(inputChannelDto);
+    when(channelService.getOutputChannelById(createStreamRequest.getOutputChannelId()))
+        .thenReturn(outputChannelDto);
+    when(streamMapper.toEntity(any())).thenReturn(streamEntity);
+    when(streamRepository.existsDuplicate(
+            createStreamRequest.getInputChannelId(), createStreamRequest.getOutputChannelId()))
+        .thenReturn(false);
+
+    // when
+    streamService.createStream(createStreamRequest);
+
+    // then
+    verify(streamRepository).save(streamEntity);
```

Comment:
I don't see any differences between this test and the next three, it doesn't seem to be asserting or checking anything? I feel like it should be verifying if the streamEntity has the right SRT mode.

Reply:
The difference is the public IP of the createStreamRequest, it increases the code coverage for the streaming mode selection. I would like to test the streaming mode, but since the returned object is an object returned by a mock, the streaming mode returned will not depend on the public IP address. An option would be to use argument captors, but I've read that can be considered a code smell. In my opinion we should use integration tests to test that behaviour.

Reply:
Since I moved that logic to `NetworkingUtil` I got rid of those tests since that logic is now tested in `NetworkingUtilTest`. Much easier to test, I should do that more often.